File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
    int n= nums.length;
    Map <Long,Integer>summap=new HashMap <>();
    long sum=0;
    int count=0;
    for(int j=0;j<n;j++){ 
        sum=0;
    for(int i=j;i<n;i++){
        sum+=nums[i];
        if(sum==k){
            count++;
        }
    }
    }return count;
    }
}

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
from collections import defaultdict
class Solution(object):
    def subarraySum(self, nums, k):
        map = defaultdict(int)
        map[0] = 1
        prefix = 0
        count = 0

        for num in nums:
            prefix += num
            if prefix - k in map:
                count += map[prefix - k]
            map[prefix] += 1

        return count        

        